source/agora/test/RestoreSlashingInfo.d

```diff
@@ -74,16 +74,23 @@ unittest
     auto b20 = set_a[0].getBlocksFrom(GenesisValidatorCycle, 1)[0];
     assert(b20.header.enrollments.length == conf.outsider_validators + 1);
 
+    // Check if the validators in the set B have all the addresses for
+    // current validators and previous validators except themselves.
+    set_b.each!(node =>
+        retryFor(node.getNodeInfo().addresses.length ==
+            GenesisValidators + conf.outsider_validators,
+            5.seconds));
+
     // Wait for nodes to run a discovery task and update their required peers
-    Thread.sleep(3.seconds);
-    // Set B validators should discover each other
-    network.waitForDiscovery();
+    Thread.sleep(conf.node.network_discovery_interval);
```

Comment:
How's that different from `waitForDiscovery` ?

Reply:
The `waitForDiscovery` can return true before the outsiders have registered again after restart.

Reply:
^ I do not quite understand why though. So I'll make a PR with just the first commit.

Reply:
This commit has been removed